src/glossary.rst

```diff
@@ -967,3 +967,117 @@ Glossary
+
+   .. code-block:: cylc
+      :caption: flow.cylc
+
+      [scheduling]
+          cycling mode = integer
+          initial cycle point = 1
+          [[graph]]
+              # repeat this for each batch
+              P1 = """
+                  # the stages of brewing in the order they must occur in
+                  malt => mash => sparge => boil => chill => ferment => rack
+
+                  # must finish the sparge of one batch before
+                  # starting the next one
+                  # sparge[-P1] => mash
+              """
+
+.. TODO - add this after the universal ID work
+
+   workflow id
+      A workflow ID is the name Cylc uses to identify a :term:`workflow` in
+      the :term:`run directory`.
+
+      This ID is used on the command line and in the GUI.
+
+      The ID is the relative path between the :term:`workflows <workflow>`
+      :term:`run directory` and the :term:`cylc run directory`.
+
+      For example, if your workflow is in ``~/cylc-run/foo/bar/run1`` then its ID
+      is ``foo/bar/run1``.
+
+      .. ..note:: For convenience you can omit the ``/run1`` on the command line.
+
+flow
+   A flow is a single logical run of a :term:`workflow` that is done by
+   a :term:`scheduler`.
+
+   A flow can be :term:`played <play>` and :term:`paused <pause>`,
+   :term:`stopped <stop>` and :term:`restarted <restart>`.
+
+   A flow begins at the :term:`start cycle point` and ends at the
+   :term:`stop cycle point`.
+
+   It is possible to run more than one flow in a single :term:`scheduler`.
+
+reflow
+   A reflow is a subsequent logical run of a :term:`workflow` that is done by
+   the same :term:`scheduler` as the original :term:`flow`.
+
+   Reflows are useful when you need to re-wind your :term:`workflow` run to
+   allow it to evolve a new path into the future.
+
+   .. TODO - add this once the CLI example works
+
+      For example, the following workflow runs the task ``a`` every cycle, then
+      either ``b`` or ``c`` depending on the outcome of ``a``:
+
+      .. code-block:: cylc
+
+         [scheduling]
+             cycling mode = integer
+             initial cycle point = 1
+             [[graph]]
+                 P1 = """
+                     b[-P1] | c[-P1] => a
+                     a:succeed => b
+                     a:fail => c
+                 """
+
+      Let's say the workflow has run to cycle 8, but we have just noticed that
+      something went wrong with the task ``a`` in cycle 5.
+
+      To rectify this we could create a new flow (a reflow) starting at ``a.5``::
+
+         # get the ID of the original flow
+         cylc TBC <id> a.5  # ???
+
+         # pause the original flow
+         cylc pause <id> --flow <flow>
+
+         # create a new flow starting at a.5
+         cylc trigger a.5 --reflow
+
+      Where ``<id>`` is the name of the :term:`workflow` and ``<flow>`` is the
+      identifier of the :term:`flow`.
+
+      This new flow will run over the top of the old one.
```

Comment:
If the old flow was *stopped*, the new flow will carry on past the point where the old flow stopped.

If the old flow was *held*(/(*paused*?), the new flow will run into the held tasks of the new flow, and the flow IDs will merge there.

Reply:
Explicitly stopping is probably un-necessary, however, a bit cleaner?

If the old and new flows take different paths a `cylc release '*'` could potentially result in the old flow continuing?

Reply:
> Explicitly stopping is probably un-necessary, however, a bit cleaner?

Yes, but easier to achieve, hence we have it already :grin: 

> If the old and new flows take different paths a cylc release '*' could potentially result in the old flow continuing?

Yes (unless you "stop" it)